extendeddatasquare.go

```diff
@@ -79,7 +90,7 @@ func ComputeExtendedDataSquare(
 func ImportExtendedDataSquare(
 	data [][]byte,
 	codec Codec,
-	treeCreatorFn TreeConstructorFn,
+	treeName string,
```

Comment:
hmmm i agreed. One question should the `ComputeExtendedDataSquare` func also be limited to these changes too since it will require another to get the treeName from the treeCreatorFn to pass in the `ExtendedDataSquare`

Reply:
i switched back to take constructorFn as argument. One problem is the way to fetch the treeName by treeConstructorFn i implemented depend on compare 2 function by using `reflect` lib, did some testing with multiple fn and i can say the method did it job but i'm still unsure. Happy to heard any other solution on this